File: DSAA Interactive Dashboard/p12.py
import os
import sys

RESET = "\033[0m"
CYAN = "\033[1;36m"
GREEN = "\033[1;32m"
RED = "\033[1;31m"
AMBER = "\033[1;33m"
PURPLE = "\033[1;35m"

def print_header(title):
    print(f"{PURPLE}\n==================================================")
    print(f"  [#] {title}")
    print(f"=================================================={RESET}")

def print_user_guide():
    print(f"{CYAN}\n==================================================")
    print("         CRASH COURSE: HOW TO USE THIS SYSTEM")
    print(f"=================================================={RESET}")
    print(" This matrix tests sorting complexities across four distinct computing behaviors:\n")
    print(f"{AMBER} [STEP 1]{RESET} Fire option 1 to allocate dynamic space and input raw registers.")
    print(f"{AMBER} [STEP 2]{RESET} Run a targeting execution strategy:")
    print(f"   • {GREEN}Bubble Sort (2):   {RESET} Sweeps elements iteratively, floating highest values (O(N^2)).")
    print(f"   • {GREEN}Selection Sort (3):{RESET} Scans looking for minimum elements, placing them at the front (O(N^2)).")
    print(f"   • {GREEN}Insertion Sort (4):{RESET} Inserts values back into a growing sorted left sub-array frame (O(N^2)).")
    print(f"   • {GREEN}Quick Sort (5):    {RESET} Logarithmic divide-and-conquer strategy using a pivot point (O(N \\log N)).")
    print(f"{PURPLE}==================================================\n{RESET}")
    input(f"Press {AMBER}[ENTER]{RESET} to initialize algorithmic sorting engine...")

def bubble(a):
    n = len(a)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]

def selection(a):
    n = len(a)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if a[j] < a[min_idx]:
                min_idx = j
        a[min_idx], a[i] = a[i], a[min_idx]

def insertion(a):
    n = len(a)
    for i in range(1, n):
        k = a[i]
        j = i - 1
        while j >= 0 and a[j] > k:
            a[j + 1] = a[j]
            j -= 1
        a[j + 1] = k

def partition(a, l, h):
    p = a[h]
    i = l - 1
    for j in range(l, h):
        if a[j] < p:
            i += 1
            a[i], a[j] = a[j], a[i]
    a[i + 1], a[h] = a[h], a[i + 1]
    return i + 1

def quick(a, l, h):
    if l < h:
        pi = partition(a, l, h)
        quick(a, l, pi - 1)
        quick(a, pi + 1, h)

def display(a):
    print(f"{GREEN}  (+) Sorted Sequence: {RESET}", end="")
    print(" ".join(map(str, a)))

def main():
    os.system('cls' if os.name == 'nt' else 'clear')
    print_user_guide()

    arr = []

    print(f"{CYAN}==================================================")
    print("     [M] ALGORITHMIC SORTING SYSTEM INTERFACE [M]")
    print(f"=================================================={RESET}")

    while True:
        print(f"{CYAN}\n [MAIN MENU] Active Matrix: {len(arr)} elements loaded{RESET}")
        print("  1.  (+) Load Data Buffer (Dynamic Array Input)")
        print("  2.  (*) Execute Bubble Sort Strategy")
        print("  3.  (*) Execute Selection Sort Strategy")
        print("  4.  (*) Execute Insertion Sort Strategy")
        print("  5.  (*) Execute Quick Sort Divide-and-Conquer")
        print("  6.  (X) Terminate Control Kernel Execution")
        
        try:
            choice = int(input(f"{AMBER} >>> Select Target Action Option: {RESET}"))
        except ValueError:
            print(f"{RED} [!] WARNING: Operational instruction unrecognized.{RESET}")
            continue

        if choice == 6:
            print(f"{RED}\n [!] CORE KERNEL POWER DOWN INTERRUPT ISSUED...\n{RESET}")
            sys.exit(0)

        if choice == 1:
            try:
                n = int(input(f"{AMBER}  [?] Array Size Limit Allocation: {RESET}"))
                if n <= 0:
                    print(f"{RED} [!] ERROR: Target index falls outside variable parameters.{RESET}")
                    arr = []
                    continue
                inp = input(f"{AMBER}  [?] Enter {n} Matrix Elements: {RESET}").split()
                arr = [int(x) for x in inp[:n]]
                print(f"{GREEN}  (+) Success: Matrix registers dynamically scaled and elements saved.{RESET}")
            except ValueError:
                print(f"{RED} [!] ERROR: Processing frame parse mismatch.{RESET}")
                arr = []
            continue

        if not arr:
            print(f"{RED} [!] ERROR: OPERATIONAL SEARCH STORAGE IS COMPLETELY EMPTY. RUN ACTION 1.{RESET}")
            continue

        if choice == 2:
            print_header("EXECUTING BUBBLE SORT STRATEGY")
            bubble(arr)
        elif choice == 3:
            print_header("EXECUTING SELECTION SORT STRATEGY")
            selection(arr)
        elif choice == 4:
            print_header("EXECUTING INSERTION SORT STRATEGY")
            insertion(arr)
        elif choice == 5:
            print_header("EXECUTING QUICK SORT DIVIDE-AND-CONQUER")
            quick(arr, 0, len(arr) - 1)
        else:
            print(f"{RED} [!] WARNING: Operational instruction unrecognized.{RESET}")
            continue

        display(arr)

if __name__ == "__main__":
    main()

File: DSAA Interactive Dashboard/p12.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <string>

#define RESET   "\033[0m"
#define CYAN    "\033[1;36m"
#define GREEN   "\033[1;32m"
#define RED     "\033[1;31m"
#define AMBER   "\033[1;33m"
#define PURPLE  "\033[1;35m"

void printHeader(const std::string& title) {
    std::cout << PURPLE << "\n==================================================\n"
              << "  [#] " << title << "\n"
              << "==================================================\n" << RESET;
}

void printUserGuide() {
    std::cout << CYAN << "\n==================================================\n"
              << "         CRASH COURSE: HOW TO USE THIS SYSTEM\n"
              << "==================================================\n" << RESET
              << " This matrix tests sorting complexities across four distinct computing behaviors:\n\n"
              << AMBER << " [STEP 1]" << RESET << " Fire option 1 to allocate dynamic space and input raw registers.\n"
              << AMBER << " [STEP 2]" << RESET << " Run a targeting execution strategy:\n"
              << "   • " << GREEN << "Bubble Sort (2):   " << RESET << " Sweeps elements iteratively, floating highest values (O(N^2)).\n"
              << "   • " << GREEN << "Selection Sort (3):" << RESET << " Scans looking for minimum elements, placing them at the front (O(N^2)).\n"
              << "   • " << GREEN << "Insertion Sort (4):" << RESET << " Inserts values back into a growing sorted left sub-array frame (O(N^2)).\n"
              << "   • " << GREEN << "Quick Sort (5):    " << RESET << " Logarithmic divide-and-conquer strategy using a pivot point (O(N \\log N)).\n"
              << PURPLE << "==================================================\n\n" << RESET
              << "Press " << AMBER << "[ENTER]" << RESET << " to initialize algorithmic sorting engine...";
    std::string dummy;
    std::getline(std::cin, dummy);
}

void bubble(std::vector<int>& a) {
    int n = a.size();
    for (int i = 0; i < n - 1; i++)
        for (int j = 0; j < n - i - 1; j++)
            if (a[j] > a[j + 1]) std::swap(a[j], a[j + 1]);
}

void selection(std::vector<int>& a) {
    int n = a.size();
    for (int i = 0; i < n - 1; i++) {
        int min = i;
        for (int j = i + 1; j < n; j++) if (a[j] < a[min]) min = j;
        std::swap(a[min], a[i]);
    }
}

void insertion(std::vector<int>& a) {
    int n = a.size();
    for (int i = 1; i < n; i++) {
        int k = a[i], j = i - 1;
        while (j >= 0 && a[j] > k) { a[j + 1] = a[j]; j--; }
        a[j + 1] = k;
    }
}

int partition(std::vector<int>& a, int l, int h) {
    int p = a[h], i = l - 1;
    for (int j = l; j < h; j++) if (a[j] < p) std::swap(a[++i], a[j]);
    std::swap(a[i + 1], a[h]);
    return i + 1;
}

void quick(std::vector<int>& a, int l, int h) {
    if (l < h) {
        int pi = partition(a, l, h);
        quick(a, l, pi - 1);
        quick(a, pi + 1, h);
    }
}

void display(const std::vector<int>& a) {
    std::cout << GREEN << "  (+) Sorted Sequence: " << RESET;
    for (int val : a) std::cout << val << " ";
    std::cout << "\n";
}

int main() {
    int choice, n = 0;
    std::vector<int> arr;

#ifdef _WIN32
    std::system("cls");
#else
    std::system("clear");
#endif

    printUserGuide();

    std::cout << CYAN << "==================================================\n"
              << "     [M] ALGORITHMIC SORTING SYSTEM INTERFACE [M]\n"
              << "==================================================\n" << RESET;

    while (true) {
        std::cout << CYAN << "\n [MAIN MENU] Active Matrix: " << n << " elements loaded\n" << RESET
                  << "  1.  (+) Load Data Buffer (Dynamic Array Input)\n"
                  << "  2.  (*) Execute Bubble Sort Strategy\n"
                  << "  3.  (*) Execute Selection Sort Strategy\n"
                  << "  4.  (*) Execute Insertion Sort Strategy\n"
                  << "  5.  (*) Execute Quick Sort Divide-and-Conquer\n"
                  << "  6.  (X) Terminate Control Kernel Execution\n"
                  << AMBER << " >>> Select Target Action Option: " << RESET;

        if (!(std::cin >> choice)) break;
        if (choice == 6) {
            std::cout << RED << "\n [!] CORE KERNEL POWER DOWN INTERRUPT ISSUED...\n" << RESET;
            break;
        }

        if (choice == 1) {
            std::cout << AMBER << "  [?] Array Size Limit Allocation: " << RESET;
            if (!(std::cin >> n) || n <= 0) {
                std::cout << RED << " [!] ERROR: Target index falls outside variable parameters.\n" << RESET;
                n = 0;
                std::cin.clear();
                std::cin.ignore(10000, '\n');
                continue;
            }
            arr.resize(n);
            std::cout << AMBER << "  [?] Enter " << n << " Matrix Elements: " << RESET;
            for (int i = 0; i < n; i++) std::cin >> arr[i];
            std::cout << GREEN << "  (+) Success: Matrix registers dynamically scaled and elements saved.\n" << RESET;
            continue;
        }

        if (n == 0) {
            std::cout << RED << " [!] ERROR: OPERATIONAL SEARCH STORAGE IS COMPLETELY EMPTY. RUN ACTION 1.\n" << RESET;
            continue;
        }

        switch (choice) {
            case 2: printHeader("EXECUTING BUBBLE SORT STRATEGY"); bubble(arr); break;
            case 3: printHeader("EXECUTING SELECTION SORT STRATEGY"); selection(arr); break;
            case 4: printHeader("EXECUTING INSERTION SORT STRATEGY"); insertion(arr); break;
            case 5: printHeader("EXECUTING QUICK SORT DIVIDE-AND-CONQUER"); quick(arr, 0, n - 1); break;
            default: std::cout << RED << " [!] WARNING: Operational instruction unrecognized.\n" << RESET; continue;
        }
        display(arr);
    }
    return 0;
}